debug_counter.h

```diff
@@ -368,17 +383,33 @@ rb_debug_counter_add(enum rb_debug_counter_type type, int add, int cond)
     return cond;
 }
 
+inline static int
+rb_debug_counter_max(enum rb_debug_counter_type type, unsigned int num)
+{
+    if (rb_debug_counter[(int)type] < num) {
+        rb_debug_counter[(int)type] = num;
+        return 1;
+    }
+    else {
+        return 0;
+    }
+}
+
 VALUE rb_debug_counter_reset(VALUE klass);
 VALUE rb_debug_counter_show(VALUE klass);
 
 #define RB_DEBUG_COUNTER_INC(type)                rb_debug_counter_add(RB_DEBUG_COUNTER_##type, 1, 1)
 #define RB_DEBUG_COUNTER_INC_UNLESS(type, cond) (!rb_debug_counter_add(RB_DEBUG_COUNTER_##type, 1, !(cond)))
 #define RB_DEBUG_COUNTER_INC_IF(type, cond)       rb_debug_counter_add(RB_DEBUG_COUNTER_##type, 1, (cond))
+#define RB_DEBUG_COUNTER_ADD(type, num)           rb_debug_counter_add(RB_DEBUG_COUNTER_##type, (num), 1)
+#define RB_DEBUG_COUNTER_SETMAX(type, num)        rb_debug_counter_max(RB_DEBUG_COUNTER_##type, (unsigned int)(num))
```

Comment:
Are we using this somewhere?

Reply:
Now they are not used, but will be used to add not incremental only counters.
For example, SETMAX is used to find maximum length of `ccs`.